sdk/rust/examples/contract_deploy.rs

```diff
@@ -0,0 +1,1596 @@
+
+    let validity = ttl_by_posix(now - 100);
+    let ttl = ttl_by_posix(now + 100);
+
+    let script = PlutusV2Script::from_bytes(hex::decode(config.lock_on).unwrap()).unwrap();
+    let contract = PlutusScript::PlutusV2(script);
+
+    let lock_on =
+        EnterpriseAddress::new(network, Credential::new_script(contract.hash())).to_address();
+    println!("Lock on contract: {}", lock_on.to_bech32(None).unwrap());
+
+    // let receipt_asset_name = if let Some(nft) = config.receipt_nft.clone() {
+    //     let mint_contract =
+    //         PlutusScript::from_hex_with_version(&nft.minting_script, &Language::new_plutus_v2())
+    //             .map_err(|err| anyhow!("Can't deserialize plutus script: {err}"))?;
+    //
+    //     let mut mint_builder = MintBuilder::new();
+    //     let asset_name = OutRef {
+    //         tx_id: config.inputs.first().cloned().unwrap().hash.to_bytes(),
+    //         index: config.inputs.first().cloned().unwrap().index,
+    //     };
+    //     let mut asset_name = PlutusData::from(asset_name).to_bytes();
+    //     asset_name.push(1);
+    //     let asset_name = blake2b_256(&asset_name).to_vec();
+    //     mint_builder.add_asset(
+    //         &MintWitness::new_plutus_script(
+    //             &PlutusScriptSource::new(&mint_contract),
+    //             &Redeemer::new(
+    //                 &RedeemerTag::new_mint(),
+    //                 &BigNum::zero(),
+    //                 &PlutusData::from(MintRedeemer::MintTokens { total: 1 }),
+    //                 &ExUnits::new(&BigNum::from(1400000u64), &BigNum::from(1000000000u64)),
+    //             ),
+    //         ),
+    //         &AssetName::new(asset_name.clone()).unwrap(),
+    //         &Int::new(&BigNum::from(1)),
+    //     );
+    //     builder.set_mint_builder(&mint_builder);
+    //
+    //     let ma = mint_builder.build().as_positive_multiasset();
+    //
+    //     builder
+    //         .add_output(
+    //             &TransactionOutputBuilder::new()
+    //                 .with_address(payment_address.clone())
+    //                 .next()
+    //                 .unwrap()
+    //                 .with_value(&Value::new_with_assets(&BigNum::from(2000000u64), &ma))
+    //                 .build()
+    //                 .unwrap(),
+    //         )
+    //         .map_err(|err| anyhow!("Can't add output: {err}"))?;
+    //     Some(asset_name)
+    // } else {
+    //     None
+    // };
+
+    let datum = match config.control_nft.clone() {
+        None if config.receipt_nft.is_none() => State {
+            owner: Owner::PKH(match payment_address.payment_cred().unwrap() {
+                StakeCredential::PubKey { hash, .. } => hash.clone(),
+                StakeCredential::Script { .. } => {
+                    return Err(anyhow!("Expected payment address, not script"))
+                }
+            }),
+            status: Status::Locked,
+        },
+        None => {
+            todo!()
```

Comment:
TODO?

Reply:
this is receipt use case, which we decided we're not spending time on